zmq/connections.nim

```diff
@@ -91,31 +91,29 @@ proc terminate*(ctx: ZContext) =
   get/set socket options
   Declare socket options first because it's used in =destroy hooks
 ]#
-# Some option take cint, int64 or uint64
-proc setsockopt_impl[T: SomeOrdinal](s: ZSocket, option: ZSockOptions, optval: T) =
-  var val: T = optval
-  if setsockopt(s, option, addr(val), sizeof(val)) != 0:
-    zmqError()
 
 # Some option take cstring
-proc setsockopt_impl(s: ZSocket, option: ZSockOptions, optval: string) =
-  var val: string = optval
-  if setsockopt(s, option, cstring(val), val.len) != 0:
-    zmqError()
+# Some option take cint, int64 or uint64
```

Comment:
@Araq The old way of using overload resolution from generic proc call doesn't work with Nim 2.2.0